### PR TITLE
Automatically detect `sdf` extension from parsed string

### PR DESCRIPTION
**Pull request by flferretti**
This PR enable detecting the XML type from the parsed string.

### CODE REVIEW

File: src/rod/sdf/sdf.py [hunk 1]
```diff
@@ -82,13 +79,12 @@ def load(sdf: pathlib.Path | str, is_urdf: bool | None = None) -> Sdf:
...
             sdf_string = pathlib.Path(sdf).read_text(encoding="utf-8")
-            is_urdf = (
-                is_urdf if is_urdf is not None else pathlib.Path(sdf).suffix == ".urdf"
-            )
+            is_urdf = pathlib.Path(sdf).suffix == ".urdf"
 
         # Finally, it must be a SDF/URDF string
         else:
             sdf_string = sdf
+            is_urdf = "sdf" not in sdf_string
```
Comment (diegoferigo) [inline src/rod/sdf/sdf.py:87]: Mmh this seems to me extremely fragile. The sequence of the three characters `s-d-f` could be part of a link name, a joint name, etc of a URDF model. A more robust solution would be looking for the string `<robot>`. This enforces the searched string to be a XML element[^1] that is specific to URDF. There is no `/robot` element supported in the SDF specification.

cc-ing @traversaro here for also having his feedback on the topic.

[^1]: I hope no one is so mentally ill to name e.g. a link as `<robot>`, which is actually a valid link name :smile:

**Review by diegoferigo — CHANGES_REQUESTED** [review]
Thanks @flferretti for this contribution! This PR introduces the automatic detection of either the URDF or SDF type when the model description is a string containing the content. Before this PR, in this case the user had to pass a `is_urdf` argument.

If the model is a `pathlib.Path`, we read its extension for the file name, and it cannot be wrong. Having some heuristics on the URDF detection also when the passed description is a string, is definitely useful. However, the proposed solution is very fragile, and I suggested an more robust solution.

Furthermore, I'd not change the APIs by removing entirely the `is_urdf` argument. With this PR, this argument will be useless in most cases, now extended to models passed as strings. Though, there might be edge cases where forcing either SDF or URDF is still necessary, and I'd suggest to leave this `is_urdf` argument there to address them. We can update the docstring to something like: *Force the SDF resource to be treated as URDF if the automatic detection fails.*

File: src/rod/sdf/sdf.py (hunk 1 above)
Comment (flferretti) [reply src/rod/sdf/sdf.py]: Thanks a lot Diego, I agree with you! Addressed in https://github.com/ami-iit/rod/pull/42/commits/4d27396c3151fc7af39b446e3ea43e37115aa5b9

**Review by diegoferigo — APPROVED** [review]
LGTM! Thanks @flferretti for your contribution.

**Pull request merged**